FirebaseRemoteConfig/Sources/FIRRemoteConfig.m

```diff
@@ -237,6 +240,61 @@ - (void)callListeners:(NSString *)key config:(NSDictionary *)config {
   }
 }
 
+- (void)setCustomSignals:(nullable NSDictionary<NSString *, NSObject *> *)customSignals
+          WithCompletion:(void (^_Nullable)(NSError *_Nullable error))completionHandler {
+  void (^setCustomSignalsBlock)(void) = ^{
+    if (!customSignals) {
+      if (completionHandler) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+          completionHandler(nil);
+        });
+      }
+      return;
+    }
+
+    for (NSString *key in customSignals) {
+      NSObject *value = customSignals[key];
+      if (value && ![value isKindOfClass:[NSString class]] &&
+          ![value isKindOfClass:[NSNumber class]]) {
+        if (completionHandler) {
+          dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Same here, any reason we have to dispatch the completion block back to main? Also, the logic here seems like, every invalid entry in dictionary will trigger completion block once, is this something we want?

Reply:
I have updated the code to use dispatch_get_global_queue here as well. 
Regarding the error handling, currently we have decided to keep the current behavior of returning an error immediately upon finding an invalid value. This provides faster feedback and simpler error handling for the caller. While accumulating all errors is an option, it could lead to delays and more complexity.